src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -223,23 +223,18 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
             return;
         }
 
-        if (evt == RETRY_WITH_H1C) {
-            // Protocol upgrade has failed, but needs to retry.
-            needsRetryWithH1C = true;
-            timeoutFuture.cancel(false);
-            ctx.close();
-            channelFactory.connect(ctx.channel().remoteAddress(), SessionProtocol.H1C, sessionPromise);
-            return;
-        }
-
         logger.warn("{} Unexpected user event: {}", ctx.channel(), evt);
     }
 
     @Override
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         failPendingResponses(ClosedSessionException.INSTANCE);
```

Comment:
Ah, forgot to mention that it does nothing but just marking the current session as inactive when `needsRetryWithH1C` is true.